maintainers/scripts/update-luarocks-packages

```diff
@@ -72,43 +70,61 @@ FOOTER="
 /* GENERATED */
 "
 
-
-function convert_pkg () {
+function convert_pkg() {
     nix_pkg_name="$1"
     lua_pkg_name="$2"
-    server="${3:+--only-server=$3}"
-    pkg_version="${4:-}"
-    lua_version="${5:+--lua-dir=$(nix path-info nixpkgs.$5)/bin}"
-
-    echo "looking at $lua_pkg_name (version $pkg_version) from server [$server]" >&2
-    cmd="luarocks nix $server $lua_version $lua_pkg_name $pkg_version"
-    echo "Running $cmd" >&2
-    drv="$nix_pkg_name = $($cmd)"
-    if [ $? -ne 0 ]; then
-        echo "Failed to convert $pkg" >&2
+    server="$3"
+    pkg_version="$4"
+    lua_version="$5"
+    maintainers="$6"
+
+    if [ "${nix_pkg_name:0:1}" == "#" ]; then
+        echo "Skipping comment ${*}" >&2
+        return
+    fi
+    if [ -z "$lua_pkg_name" ]; then
+        echo "Using nix_name as lua_pkg_name for '$nix_pkg_name'" >&2
+        lua_pkg_name="$nix_pkg_name"
+    fi
+
+    echo "Building expression for $lua_pkg_name (version $pkg_version) from server [$server]" >&2
+    luarocks_args=(nix)
+    if [[ -n $server ]]; then
+        luarocks_args+=("--only-server=$server")
+    fi
+    if [[ -n $maintainers ]]; then
+        luarocks_args+=("--maintainers=$maintainers")
+    fi
+    if [[ -n $lua_version ]]; then
+        luarocks_args+=("--lua-dir=$(nix path-info "nixpkgs.$lua_version")/bin")
```

Comment:
Here we would need a code that realizes the derivation, like nix-build -A . Otherwise the path might not exist.

Reply:
Thats true. `nix-build` itself returns the derivation store path anyway, so I think it could just use it instead of path-info. I'll give that a try in a bit.